crates/pyhq/python/hyperqueue/task/function.py

```diff
@@ -13,19 +13,32 @@ def purge_cache():
     _PICKLE_CACHE = {}
 
 
-def cloud_wrap(fn, cache=True):
+def cloud_wrap(fn, cache=True) -> CloudWrapper:
     if isinstance(fn, CloudWrapper):
         return fn
     return CloudWrapper(fn, cache=cache)
 
 
+def task_main():
+    import pickle
+    import sys
+
+    try:
+        fn, a, kw = pickle.loads(sys.stdin.buffer.read())
+        fn(*a, **(kw if kw is not None else {}))
+    except Exception:
+        import os
+        import traceback
+
+        t = traceback.format_exc()
+        with open(os.environ["HQ_ERROR_FILENAME"], "w") as f:
```

Comment:
If `task_dir` is false, then this environment variable won't exist?

Reply:
It is true, but user cannot create Python task without task_dir